app/src/laos/java/org/eyeseetea/malariacare/domain/usecase/LoginUseCase.java

```diff
@@ -0,0 +1,68 @@
+package org.eyeseetea.malariacare.domain.usecase;
+
+import org.eyeseetea.malariacare.domain.boundary.IAuthenticationManager;
+import org.eyeseetea.malariacare.domain.entity.Credentials;
+import org.eyeseetea.malariacare.domain.entity.UserAccount;
+import org.eyeseetea.malariacare.domain.exception.InvalidCredentialsException;
+import org.eyeseetea.malariacare.domain.exception.NetworkException;
+
+import java.net.MalformedURLException;
+import java.net.UnknownHostException;
+
+public class LoginUseCase extends ALoginUseCase {
+    private IAuthenticationManager mAuthenticationManager;
+
+    public LoginUseCase(IAuthenticationManager authenticationManager) {
+        mAuthenticationManager = authenticationManager;
+    }
+
+    @Override
+    public void execute(final Credentials credentials, final Callback callback) {
```

Comment:
@idelcano this code is hard to read, can you refactor to more readable code creating methods with semantic name instead of indented callbacks?